### ISSUE TITLE
Invalid selector generated for psuedo-elements with default config

### ISSUE THREAD

**Issue by iranana** [Roman86/tailwindcss-scoped-preflight#58]
**Describe the bug**
When using the default settings for the `isolateInsideOfContainer` strategy, an invalid selector is generated for pseudo-elements, e.g:

```
*:where(#somecontainer,#somecontainer *),
::before:where(#somecontainer,#somecontainer *),
::after:where(#somecontainer,#somecontainer *) {
  box-sizing: border-box; /* 1 */
  border-width: 0; /* 2 */
  border-style: solid; /* 2 */
  border-color: #e5e7eb; /* 2 */
}

::before:where(#somecontainer,#somecontainer *),
::after:where(#somecontainer,#somecontainer *) {
  --tw-content: '';
}
```

Apparently, not all pseudo-classes can be compounded with a pseudo-element. This appears to include `:where`:
https://www.w3.org/TR/selectors/#pseudo-element-states

It seems that as of now, the only pseudo-classes that should be considered valid are the following:
https://www.w3.org/TR/selectors/#useraction-pseudos

As a result, Tailwind's utilities for pseudo-elements as outlined below do not work as expected, especially when not specifying `content-`:
https://tailwindcss.com/docs/hover-focus-and-other-states#pseudo-elements

**Expected behaviour**
There are workarounds (e.g. using `remove` or `ignore` or adding `content-`), but I didn't see anything in the documentation about this behaviour, so it would be good to highlight it there. Maybe this particular preflight style could also be handled differently by default, but I'm not really sure what the best option is there.

**Screenshots**
Given the following, we expect a blue square:
`<div class="after:block after:size-12 after:bg-blue-500">should have blue square after</div>`

With `ignore: ["::before", "::after"]`:
![image](https://github.com/user-attachments/assets/7feacdb8-8f6c-4017-955f-324bc963a92c)

Default settings (without `ignore`):
![image](https://github.com/user-attachments/assets/0630198a-77a7-4be1-b1c6-9a3791847654)

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I have identified the issue with the invalid selector generation for pseudo-elements in the `isolateInsideOfContainer` strategy. The problem was that the `:where` pseudo-class was being incorrectly combined with pseudo-elements like `::before` and `::after`. To fix this, I've modified the `isolateInsideOfContainer` function in the `src/strategies.ts` file to handle pseudo-elements separately.

Here's a summary of the changes:

1. Added a function to detect pseudo-elements.
2. Modified the `isolateInsideOfContainer` function to handle pseudo-elements differently.
3. Updated the selector generation logic to ensure proper handling of pseudo-elements.

These changes should resolve the issue with invalid selectors for pseudo-elements while maintaining the existing functionality for other selectors.

I also created pull request: https://github.com/Roman86/tailwindcss-scoped-preflight/pull/59


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

**Comment by Roman86** [comment]
hi @iranana and thanks for reaching out! Had no time to dive into it yet. Will be carefully reviewed and fixed. Stay tuned

**Comment by Roman86** [comment]
hi again @iranana !
Fixed. I tested the case you provided - it's good now. Although tests pass well, please check the beta version and confirm that there is no any degradations on your side:

```sh
npm i tailwindcss-scoped-preflight@beta
```
```sh
npm i tailwindcss-scoped-preflight@3.4.7
```

**Comment by iranana** [comment]
@Roman86 thank you for all your work on this library :smile:

**Comment by Roman86** [comment]
hi @iranana ! My fix caused another issue with pseudo-elements: #60
Just fixed it. The `latest` tag temporarily moved back to `v3.4.6`. Please try the latest fix.

```sh
npm i tailwindcss-scoped-preflight@3.4.9
```
or
```sh
npm i tailwindcss-scoped-preflight@beta
```
This fixed versions is not yet tagged as `latest`. Until confirmed.
